File: euler-number/c/simple_euler.c
/*
Heru Handika
12 September 2020
Find Euler's number using factorials
*/

#include <stdio.h>
#include <stdlib.h>

int calculate_factorial(int n);
double calculate_euler_number(int n);

int main(void) {
    int n = 10;
    double sum = calculate_euler_number(n);
    printf("%.8lf\n", sum);

    return EXIT_SUCCESS;
}

int calculate_factorial(int n) {
    if (n == 0 || n == 1) return 1;
    else {
        return n * calculate_factorial(n - 1);
    }     
}

double calculate_euler_number(int n) {
    double sum;
    int numbers = 1;
    while (numbers < n) {
        sum += 1.0 / (double) calculate_factorial(numbers);
        numbers++;
    }
    return sum + 1;
}

File: euler-number/python/simple_euler.py
"""
Heru Handika
12 September 2020
Euler's number finder.
Using for loop
"""
#%%
def calculate_factorial(n: int) -> int:
    if n == 0 or n == 1:
        return 1
    else:
        return n * calculate_factorial(n-1)

def calculate_euler(n: int) -> int:
    return 1 + sum((1.0/calculate_factorial(i)) for i in range(1, n))

def main():
    euler_num: float = calculate_euler(10)
    print(f"Euler's number using vectorization {euler_num: .4f}")

if __name__ == "__main__":
    main()